ETL_pipeline/extract.py

```diff
@@ -0,0 +1,46 @@
+"""This file fetches weather information for various locations."""
+from datetime import datetime, timedelta
+from os import environ as ENV
+
+from dotenv import load_dotenv
+import requests
+
+
+def get_weather_details_for_week(latitude: int, longitude: int) -> dict:
+    """This function extract the weather details for the coming week."""
+    today = datetime.now()
+    next_week = today + timedelta(weeks=1)
+    today_str = today.isoformat(timespec='minutes')
+    next_week_str = next_week.isoformat(timespec='minutes')
+    response = requests.get(
+        f'https://api.open-meteo.com/v1/forecast?latitude={str(latitude)}&longitude={str(longitude)}&hourly=apparent_temperature,cloud_cover,relative_humidity_2m,lightning_potential,precipitation,precipitation_probability,rain,snowfall,temperature_2m,uv_index,visibility,wind_direction_10m,wind_gusts_10m,wind_speed_10m,weather_code&start_hour={today_str}&end_hour={next_week_str}',
+        timeout=3)
+    return response.json()
+
+
+def get_weather_details_for_24hrs(latitude: int, longitude: int) -> dict:
+    """This function extracts the weather details for the next 24hours."""
+    today = datetime.now()
+    tomorrow = today + timedelta(days=1)
+    today_str = today.isoformat(timespec='minutes')
+    tomorrow_str = tomorrow.isoformat(timespec='minutes')
+    response = requests.get(
+        f'https://api.open-meteo.com/v1/forecast?latitude={str(latitude)}&longitude={str(longitude)}&hourly=cloud_cover,apparent_temperature,relative_humidity_2m,lightning_potential,precipitation,precipitation_probability,rain,snowfall,temperature_2m,uv_index,visibility,wind_direction_10m,wind_gusts_10m,wind_speed_10m,weather_code&start_hour={today_str}&end_hour={tomorrow_str}',
```

Comment:
This is the same string as before? I think you meant 'minutely_15' where you have 'hourly', and start_minutely_15/end_minutely_15. I think you copied the wrong string, use this: f'https://api.open-meteo.com/v1/forecast?latitude={str(latitude)}&longitude={str(longitude)}&minutely_15=apparent_temperature,cloud_cover,relative_humidity_2m,lightning_potential,precipitation,precipitation_probability,rain,snowfall,temperature_2m,uv_index,visibility,wind_direction_10m,wind_gusts_10m,wind_speed_10m,weather_code&start_minutely_15={today_str}&end_minutely_15={tomorrow_str}'